cmd/application.go

```diff
@@ -47,14 +47,31 @@ var applicationCreateCmd = &cobra.Command{
 	Example: `  # Create an application
   odo app create myapp
 	`,
-	Args: cobra.ExactArgs(1),
+	Args: cobra.RangeArgs(0, 1),
 	Run: func(cmd *cobra.Command, args []string) {
-		// Args validation makes sure that there is exactly one argument
-		name := args[0]
+		var name string
+		client := getOcClient()
+		if len(args) == 1 {
+			// The only arg passed is the app name
+			name = args[0]
+		} else {
+			// Desired app name is not passed so, generate a new app name
+			// Fetch existing list of apps
+			apps, err := application.List(client)
+			checkError(err, "")
+			appNames := []string{}
```

Comment:
Since we are forming this list for searching later only, we can switch to a `map[string]bool`. It will make the searching faster and a lot easier and we won't need the StringInSlice() function too. The search will become a single line like map["app_name"], if it returns `nil`, then the app_name is not used.

Reply:
Agreed. But though we have unit tests for StringInSlice(), it will still add the necessity to maintain the function. Also there is the efficiency thing. Anyways I will leave it up to you.

Reply:
I would take opinion from @kadel before doing this because I remember receiving a comment from him on my imagestreams PR in a similar scenario where I modified the list, search etc... functions to  use maps instead of slice of objects, but I reverted it to same old state of slice of objects instead of maps with readability as the main reason

Reply:
Another way might be to convert the list to a map inside the GetRandomName() function, which will take one iteration and then performing the search on the map. WDYT?

Reply:
I think that sounds good